web-registry/src/lib/bank.ts

```diff
@@ -0,0 +1,160 @@
+import {
+  QueryClientImpl,
+  DeepPartial,
+  QueryBalanceRequest,
+  QueryBalanceResponse,
+  QueryDenomMetadataRequest,
+  QueryDenomMetadataResponse,
+  QueryAllBalancesRequest,
+  QueryAllBalancesResponse,
+  QueryDenomsMetadataRequest,
+  QueryDenomsMetadataResponse,
+} from '@regen-network/api/lib/generated/cosmos/bank/v1beta1/query';
+
+// TODO: pagination not implemented yet
+
+/**
+ *
+ * BANK MODULE QUERIES
+ * -------------------
+ *  - AllBalances
+ *  - Balance
+ *  - DenomMetadata
+ *  - DenomsMetadata
+ *
+ */
+
+/**
+ *
+ * QUERY TYPES
+ *
+ */
+
+// typing the query client
+
+export type BankQueryClient = QueryClientImpl;
+
+interface BankQueryClientProps {
+  client: BankQueryClient;
+}
+
+// typing and linking query names and corresponding input params
+
+type AllBalancesParams = {
+  query: 'allBalances';
+  params: DeepPartial<QueryAllBalancesRequest>;
+};
+
+type BalanceParams = {
+  query: 'balance';
+  params: DeepPartial<QueryBalanceRequest>;
+};
+
+type DenomMetadataParams = {
+  query: 'denomMetadata';
+  params: DeepPartial<QueryDenomMetadataRequest>;
+};
+
+type DenomsMetadataParams = {
+  query: 'denomsMetadata';
+  params: DeepPartial<QueryDenomsMetadataRequest>;
+};
+
+export type BankQueryProps =
+  | AllBalancesParams
+  | BalanceParams
+  | DenomMetadataParams
+  | DenomsMetadataParams;
+
+// TODO ?
+// typing the response
+// export type BankQueryResponse = QueryDenomMetadataResponse | QueryBalanceResponse;
+
+/**
+ *
+ * QUERY FUNCTIONS
+ *
+ */
+
+// All Balances
+
+interface QueryAllBalancesProps extends BankQueryClientProps {
+  request: DeepPartial<QueryAllBalancesRequest>;
+}
+
+export const queryAllBalances = async ({
+  client,
+  request,
+}: QueryAllBalancesProps): Promise<DeepPartial<QueryAllBalancesResponse>> => {
+  try {
+    return await client.AllBalances({
+      address: request.address,
+    });
+  } catch (err) {
+    throw new Error(
+      'Error in the AllBalances query of the ledger bank module.',
```

Comment:
I think it'd be good to actually use the `err` in `new Error(...)`, otherwise the error msg is not super helpful
ditto for all the similar error handling

Reply:
Seem right. Just concatenating the `err` to the string, right?